spyderlib/spyder.py

```diff
@@ -2660,6 +2677,87 @@ def show_tour(self, index):
         self.tour.set_tour(index, frames, self)
         self.tour.start_tour()
 
+    # ---- Check for Spyder Updates
+    def _check_updates_ready(self, feedback):
+        """Called by WorkerUpdates when ready"""
+        from distutils.version import StrictVersion as Version
+
+        version = __version__
+        latest_release = self.worker_updates.latest_release
+        if DEV is None:
+            # In case someone installs a release candidate in which this might
+            # fail.
+            try:
+                update_available = Version(latest_release) > Version(version)
+            except Exception:
+                update_available = False
+        else:
+            update_available = False
+
+        if update_available:
+            msg = _("Spyder {0} is available.".format(latest_release))
+            QMessageBox.information(self, _("Update available"), msg)
+        elif feedback:
+            msg = _("Spyder is up to date.")
+            QMessageBox.information(self, _("No updates available"), msg)
+
+        # Enable check_updates_action
+        self.check_updates_action.setDisabled(False)
+
+    def check_updates(self, feedback):
+        """
+        Check for spyder updates on github releases using a QThread.
+
+        `feedback` = False is used on startup, so only positive (updates found)
+        feedback is given. `feedback` = True is used when using the menu, and
+        gives feeback if updates are, or are not found.
+        """
+        # Disable check_updates_action
+        self.check_updates_action.setDisabled(True)
+        if self.thread_updates is not None:
+            self.thread_updates.terminate()
+        self.thread_updates = QThread(self)
+        self.worker_updates = WorkerUpdates(self, feedback)
+        self.worker_updates.sig_ready.connect(self._check_updates_ready)
+        self.worker_updates.sig_ready.connect(self.thread_updates.quit)
+        self.worker_updates.moveToThread(self.thread_updates)
+        self.thread_updates.started.connect(self.worker_updates.check)
+        self.thread_updates.start()
+
+
+class WorkerUpdates(QObject):
+    """
+    Worker that checks for releases using the Github API without blocking the
+    Spyder user interface.
+    """
+    sig_ready = Signal(bool)
+
+    def __init__(self, parent, feedback):
+        QObject.__init__(self)
+        self._parent = parent
+        self._feedback = feedback
+        self.latest_release = None
+
+    def check(self):
+        """ """
+        import json
+        try:
+            from urllib.request import urlopen
+        except ImportError:
+            from urllib import urlopen
+
+        url = 'https://api.github.com/repos/spyder-ide/spyder/releases'
+
+        try:
+            page = urlopen(url)
+            data = json.loads(page.read())
+            self.latest_release = data[0]['tag_name'].replace('v', '')
+        except Exception:

```

Comment:
I don't know, what exceptions can occur here? I mean, is not really good practice to except on everything.